types/wordpress__block-editor/components/block-format-controls.d.ts

```diff
@@ -1,4 +1,4 @@
-import { Slot } from "@wordpress/components";
+import { SlotComponentProps } from "@wordpress/components/build-types/slot-fill/types";
```

Comment:
Just one nit, I don't know if accessing the types this way is safe. Specifically, whether Gutenberg's public API rules would indicate that this is a _stable_ way to reference these types. It might create a hard-pin on a specific version that wouldn't be obvious (usually moving files around isn't cause for even a patch version bump in semver).

Reply:
This is the only option I was aware of. The types are not exported from the main package file.
Maybe @gziolo knows the answer?

Reply:
I did it differently on my PR, I used Parameters to avoid this issue, the only type I had to import like this was Popover. Perhaps they should be imported that way instead? Wordpress doesn't explicitly export its types for Props, only for the full components.
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69143

Reply:
The list of props for React components definitely sounds like public API. I have no insights why would they be not exposed from the root of the package.

Reply:
Unfortunately, in your approach, for example, the `typeof slot` is of type `unknown`. I tried with `React.ComponentProps<typeof Slot>`, but also the result is `unkwown`.
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/25438601/65f5a53a-e9b5-459e-9ae1-aa49d9cad373)

Direct import of types:
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/25438601/7c6e2695-0d23-4c4e-874c-35ccd6390f72)

Reply:
Ah I didn't notice that, well spotted

Reply:
Issue submitted: https://github.com/WordPress/gutenberg/issues/60193

Reply:
So the plan is:

1. Merge this PR as it is
2. Create PR to @wordpress/components that will export types from root
3. Wait for merge and release
4. Create second PR to DT that will update imports

Do you agree @sarayourfriend?

Reply:
Sounds great!

Reply:
Yes, sounds good to me.

Reply:
Ok. Can you merge this PT then?

Reply:
Merging this is blocked on:

> This PR touches some part of DefinitelyTyped infrastructure, so a DT maintainer will need to review it. This is rare — did you mean to do this?

And from the checklist:

> A DT maintainer needs to approve changes which affect DT infrastructure ([notNeededPackages.json](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69133/files/af6acf82514e343c8d5d16e4752317a97e12c892#diff-a275c2eae7b8f788a52327e76809026ee1bc5ea614393806c8f3cbab07202621))

It's been a while since I've done this, so I'd check out the contributing documentation for the repo to see if there are instructions on getting that review from a DT maintainer :+1:

Reply:
The documentation does not say exactly how to do this. There is only [information](https://github.com/DefinitelyTyped/dt-mergebot/blob/master/docs/dt-mergebot-lifecycle.svg) that the maintainer must accept it.
I see from other pull requests that some of the maintainers are @peterblazejewicz, @sheetalkamat.

Reply:
Okay, well that graph makes it look like with my approval this PR will be in the DT maintainer review queue, so just hold tight while they get to it, I guess.